packages/ui/src/components/notifications/NotificationCenter.tsx

```diff
@@ -0,0 +1,88 @@
+import { ComponentType, useEffect, useState } from 'react';
+
+import { Notification } from './types';
+
+const NOTIFICATION_EVENT = 'emit-core-notification';
+const DEFAULT_CACHE_TIMEOUT = 5 * 60 * 1000; // 5 minutes in milliseconds
+const DEFAULT_MAX_NOTIFICATIONS = 5;
+
+export const notify = (props: Omit<Notification, 'id' | 'createdAt'>) => {
+    const id = window.crypto.randomUUID();
+    const createdAt = new Date();
+    const event = new CustomEvent(NOTIFICATION_EVENT, {
+        detail: { id, open: true, createdAt, ...props },
+    });
+    document.dispatchEvent(event);
+};
+
+type SnackbarComponentProps = Notification & {
+    onClose: (id: string) => void;
+};
+
+type NotificationCenterProps = {
+    maxNotifications?: number;
+    hideAfter?: number;
+    SnackbarComponent: ComponentType<SnackbarComponentProps>;
+    cacheTimeout?: number;
+    className?: string;
+};
+
+const NotificationCenter = ({
```

Comment:
In order to be imported into the client apps, this needs to be re-exported in `packages/ui/src/index.ts` like the table component. You can make a barrel export for things inside the /notification folder where you export the component itself and the types